File: app/src/main/java/com/nptel/courses/online/utility/ShimmerViewConverter.java
package com.nptel.courses.online.utility;

import android.view.View;
import android.view.ViewGroup;

import com.facebook.shimmer.ShimmerFrameLayout;
import com.nptel.courses.online.R;

public class ShimmerViewConverter {

    public static View wrap(View view) {
        ShimmerFrameLayout shimmerFrameLayout = new ShimmerFrameLayout(view.getContext());
        ViewGroup.LayoutParams layoutParams = new ViewGroup.LayoutParams(view.getLayoutParams());

        shimmerFrameLayout.setLayoutParams(layoutParams);

        shimmerFrameLayout.addView(view);
        shimmerFrameLayout.startShimmer();
        return shimmerFrameLayout;
    }

    public static View convert(View view) {

        ShimmerFrameLayout shimmerFrameLayout = new ShimmerFrameLayout(view.getContext());
        ViewGroup.LayoutParams layoutParams = new ViewGroup.LayoutParams(view.getLayoutParams());

        shimmerFrameLayout.setLayoutParams(layoutParams);
        if (view instanceof ViewGroup) setBackground((ViewGroup) view);
        else
            view.setBackgroundColor(view.getContext().getResources().getColor(R.color.mildBackgroundColor));
        shimmerFrameLayout.addView(view);
        shimmerFrameLayout.startShimmer();
        return shimmerFrameLayout;
    }

    static void setBackground(ViewGroup view) {
        for (int i = 0; i < view.getChildCount(); i++)
            if (view.getChildAt(i) instanceof ViewGroup)
                setBackground((ViewGroup) view.getChildAt(i));
            else
                view.getChildAt(i).setBackgroundColor(view.getContext().getResources().getColor(R.color.mildBackgroundColor));

    }
}

File: app/src/main/java/com/nptel/courses/online/utility/ShimmerViewConverter.kt
package com.nptel.courses.online.utility

import android.view.View
import android.view.ViewGroup
import com.facebook.shimmer.ShimmerFrameLayout
import com.nptel.courses.online.R

object ShimmerViewConverter {
    fun wrap(view: View): View {
        val shimmerFrameLayout = ShimmerFrameLayout(view.context)
        val layoutParams = ViewGroup.LayoutParams(view.layoutParams)
        shimmerFrameLayout.layoutParams = layoutParams
        shimmerFrameLayout.addView(view)
        shimmerFrameLayout.startShimmer()
        return shimmerFrameLayout
    }

    fun convert(view: View): View {
        val shimmerFrameLayout = ShimmerFrameLayout(view.context)
        val layoutParams = ViewGroup.LayoutParams(view.layoutParams)
        shimmerFrameLayout.layoutParams = layoutParams
        if (view is ViewGroup) setBackground(view) else view.setBackgroundColor(view.context.resources.getColor(R.color.mildBackgroundColor))
        shimmerFrameLayout.addView(view)
        shimmerFrameLayout.startShimmer()
        return shimmerFrameLayout
    }

    fun setBackground(view: ViewGroup) {
        for (i in 0 until view.childCount) if (view.getChildAt(i) is ViewGroup) setBackground(view.getChildAt(i) as ViewGroup) else view.getChildAt(i).setBackgroundColor(view.context.resources.getColor(R.color.mildBackgroundColor))
    }
}